internal/commands/vaultsecrets/apps/read.go

```diff
@@ -0,0 +1,86 @@
+package apps
+
+import (
+	"context"
+	"fmt"
+
+	preview_secret_service "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/client/secret_service"
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/stable/2023-06-13/client/secret_service"
+	"github.com/hashicorp/hcp/internal/pkg/cmd"
+	"github.com/hashicorp/hcp/internal/pkg/format"
+	"github.com/hashicorp/hcp/internal/pkg/heredoc"
+	"github.com/hashicorp/hcp/internal/pkg/iostreams"
+	"github.com/hashicorp/hcp/internal/pkg/profile"
+)
+
+type ReadOpts struct {
+	Ctx     context.Context
+	Profile *profile.Profile
+	Output  *format.Outputter
+	IO      iostreams.IOStreams
+
+	AppName       string
+	Description   string
+	Client        secret_service.ClientService
+	PreviewClient preview_secret_service.ClientService
+}
+
+func NewCmdRead(ctx *cmd.Context, runF func(*ReadOpts) error) *cmd.Command {
+	opts := &ReadOpts{
+		Ctx:           ctx.ShutdownCtx,
+		Profile:       ctx.Profile,
+		Output:        ctx.Output,
+		IO:            ctx.IO,
+		Client:        secret_service.New(ctx.HCP, nil),
+		PreviewClient: preview_secret_service.New(ctx.HCP, nil),
+	}
+
+	cmd := &cmd.Command{
+		Name:      "read",
+		ShortHelp: "Read a Vault Secrets application.",
+		LongHelp: heredoc.New(ctx.IO).Must(`
+		The {{ template "mdCodeOrBold" "hcp vault-secrets apps read" }} command gets a Vault Secrets application.
+		`),
+		Examples: []cmd.Example{
+			{
+				Preamble: `Read an application:`,
+				Command: heredoc.New(ctx.IO, heredoc.WithPreserveNewlines()).Must(`
+				$ hcp vault-secrets apps read company-card
+				`),
+			},
+		},
+		Args: cmd.PositionalArguments{
+			Args: []cmd.PositionalArgument{
+				{
+					Name:          "APP_NAME",
```

Comment:
not sure, but should we keep this as NAME too?

Reply:
I think it's APP_NAME across vaultsecrets/apps. Were you referring to `NAME` somewhere else?

Reply:
I was referring to this comment - https://hashicorp.slack.com/archives/C0630MA5H8B/p1715722134440319?thread_ts=1715721973.724659&cid=C0630MA5H8B

Reply:
Gotcha, will change them